specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2020-04-01-preview/keyvault.json

```diff
@@ -1939,6 +1943,59 @@
         }
       },
       "description": "Properties of a private link resource."
+    },
+    "SystemData": {
+      "description": "Metadata pertaining to creation and last modification of key vault resource.",
+      "type": "object",
+      "readOnly": true,
+      "properties": {
+        "createdBy": {
+          "type": "string",
+          "description": "The identity that created key vault resource."
+        },
+        "createdByType": {
+          "type": "string",
+          "description": "The type of identity that created key vault resource.",
+          "enum": [
+            "User",
+            "Application",
+            "ManagedIdentity",
+            "Key"
+          ],
+          "x-ms-enum": {
+            "name": "createdByType",
+            "modelAsString": true
+          }
+        },
+        "createdAt": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The timestamp of key vault resource creation (UTC)."
+        },
+        "lastModifiedBy": {
+          "type": "string",
+          "description": "The identity that last modified key vault resource."
+        },
+        "lastModifiedByType": {
+          "type": "string",
+          "description": "The type of identity that last modified key vault resource.",
+          "enum": [
+            "User",
+            "Application",
+            "ManagedIdentity",
+            "Key"
+          ],
+          "x-ms-enum": {
+            "name": "createdByType",
```

Comment:
This will cause conflicts during generation. Instead, I recommend refactoring out this enum as a separate definition and `#ref` it in both places (here and `createdByType`).

Reply:
Sorry, there is a copy/paste error. I have refactored out the identity type.